crates/ra_hir_expand/src/db.rs

```diff
@@ -75,6 +88,8 @@ pub trait AstDatabase: SourceDatabase {
 
     #[salsa::interned]
     fn intern_eager_expansion(&self, eager: EagerCallLoc) -> EagerMacroId;
+
+    fn expand_proc_macro(&self, call: ProcMacroCall) -> Result<tt::Subtree, mbe::ExpandError>;
```

Comment:
Hm, so this stores `TokenMap` as a **key**, and that's why we need all those `Hash` impls. I think we don't want to do this, because we'll efectively have a separate key for each change to the derive-annotated code, and then we'll need to gc it....

It should be sufficient to use `MacroCallId` or something similar as they key.

Reply:
Agreed and fixed. And please note that we currently don't gc `hir::db::macro_arg` such that the `TokenMap` is not gc too. But I agreed we should do it in the future.